### PR TITLE
implemented issue #33 and a stub for the backend

### PR DESCRIPTION
**Pull request by johannesfbraun**

### CODE REVIEW

**Review by KevinGus — APPROVED** [review]
LGTM

File: client/src/ts/members/ChangePassword.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+          helperText=" Altes Passwort ist nicht korrekt" value={oldPassword}
+          onChange={event => { setOldPassword(event.target.value); }} fullWidth />
+      );
+    }
+    return (
+      <Textfield className={classes.inputfield} id="password" label="altes Passwort" type="password" value={oldPassword}
+        onChange={event => { setOldPassword(event.target.value); }} fullWidth />
+    );
+  };
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:126]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+          fullWidth />
+      );
+    } else {
+      return (
+        <Textfield className={classes.inputfield} id="newpassword" label="neues Passwort" type="password" value={newPassword}
+          onChange={event => { setNewPassword(event.target.value); }}
+          fullWidth />
+      );
+    }
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:145]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    }
+    else {
+      return null;
+    }
+  };
+
+  return (
+    <div>
+      <div className="content-page">
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:199]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        <Paper className={classes.paper}>
+          <p>
+            Das neue Passwort muss mindestens 8 Zeichen lang sein und eine Zahl, einen kleinen- und einen großen Buchstaben enthalten
+          </p>
+        </Paper>
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:205]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        <Paper className={classes.paper}>
+          <div>
+            {getOldPasswordField()}
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:209]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+            {getNewPasswordField()}
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:211]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+              Neues Passwort speichern
+          </Button>
+          </div>
+          <div>
+            <p>
+              {postNotSentWarning()}
+            </p>
+          </div>
+        </Paper>
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:225]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+        {resResponse200Field()}
+
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:227]: Delete empty line.

File: client/src/ts/global/App.tsx [hunk 9]
```diff
@@ -8,6 +8,8 @@ import MemberOverview from "../members/MemberOverview";
 import Login from "../members/Login";
 import Nav from "./navigation/Nav";
 import NotFound from "./NotFound";
+import ChangePassword from "../members/ChangePassword";
+
```
Comment (LennartRukower) [inline client/src/ts/global/App.tsx:12]: Delete empty line.

File: server/src/members/membersController.ts [hunk 10]
```diff
@@ -5,78 +5,86 @@ import database = require("../database");
...
+};
+
+/**
+ * User can change his password knowing his old one
+ */
+export const changePassword = (req: Request, res: Response): void => {
+  res.status(200).send("not implemented yet");
 };
 
+
```
Comment (LennartRukower) [inline server/src/members/membersController.ts:62]: Delete empty line.

File: client/src/ts/members/ChangePassword.tsx [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+      })
+        .then((res) => {
+          if (res.status === 200) {
+            // Worked
+            setResResponse200(true);
+            return [{ status: res.status }];
+          } else {
+            return [{ status: res.status }];
+            setFailedoldPassword(true);
+            setResResponse200(false);
```
Comment (LennartRukower) [inline client/src/ts/members/ChangePassword.tsx:84]: Both lines (83 and 84) will never be executed.

**Review by LennartRukower — CHANGES_REQUESTED** [review]

**Review by LennartRukower — APPROVED** [review]

**Pull request merged**